src/main/java/com/github/danielwegener/logback/kafka/KafkaAppenderConfig.java

```diff
@@ -21,43 +48,66 @@
 
     protected String topic = null;
 
-    protected KafkaMessageEncoder<E> encoder = null;
+    protected Encoder<E> encoder = null;
     protected KeyingStrategy<? super E> keyingStrategy = null;
     protected DeliveryStrategy deliveryStrategy;
 
     public static final Set<String> KNOWN_PRODUCER_CONFIG_KEYS = new HashSet<String>();
     public static final Map<String,String> DEPRECATED_PRODUCER_CONFIG_KEYS = new HashMap<String, String>();
     static {
         KNOWN_PRODUCER_CONFIG_KEYS.add(BOOTSTRAP_SERVERS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(METADATA_FETCH_TIMEOUT_CONFIG);
-        DEPRECATED_PRODUCER_CONFIG_KEYS.put(METADATA_FETCH_TIMEOUT_CONFIG, MAX_BLOCK_MS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(METADATA_MAX_AGE_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(BATCH_SIZE_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(BUFFER_MEMORY_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(ACKS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(TIMEOUT_CONFIG);
-        DEPRECATED_PRODUCER_CONFIG_KEYS.put(METADATA_FETCH_TIMEOUT_CONFIG, REQUEST_TIMEOUT_MS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(LINGER_MS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(CLIENT_ID_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(SEND_BUFFER_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(RECEIVE_BUFFER_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(MAX_REQUEST_SIZE_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(RECONNECT_BACKOFF_MS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(BLOCK_ON_BUFFER_FULL_CONFIG);
-        DEPRECATED_PRODUCER_CONFIG_KEYS.put(METADATA_FETCH_TIMEOUT_CONFIG, null);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(RETRIES_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(RETRY_BACKOFF_MS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(COMPRESSION_TYPE_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(METRICS_SAMPLE_WINDOW_MS_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(METRICS_NUM_SAMPLES_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(METRIC_REPORTER_CLASSES_CONFIG);
-        KNOWN_PRODUCER_CONFIG_KEYS.add(MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION);
         KNOWN_PRODUCER_CONFIG_KEYS.add(KEY_SERIALIZER_CLASS_CONFIG);
         KNOWN_PRODUCER_CONFIG_KEYS.add(VALUE_SERIALIZER_CLASS_CONFIG);
+        KNOWN_PRODUCER_CONFIG_KEYS.add(ACKS_CONFIG);
```

Comment:
IMHO Checking which Kafka properties are valid should be the reponsibility of the KafkaProducer not the KafkaAppender.
Does it prevent depending on newer Kafka client and using new config keys, like sasl.jaas.config config added in Kafka 0.10.2 for example?
To make it more dynamic, this Set could be filled using reflection on the ProducerConfig class and looking for static fields of type String suffixed "_CONFIG".

Reply:
Agreed. I came along this one earlier. Essentially it would be nice if we could access the `ProducerConfig.Config`object and check if it understands the given properties. But unfortunately it seems to be internal api and I do not want to implement against internal api that may change at any time. I'd rather tend to remove the check.